File: Julia/monte_carlo_pi_estimation.jl
function monte_carlo_pi_estimation(num_points::Int)
    # Estimate Pi using Monte Carlo Simulation
    # Args:
    #     num_points: Number of random points to generate.
    # Returns:
    #     pi_estimate: Estimated value of Pi.
    
    inside_circle = 0

    for i in 1:num_points
        x, y = rand(), rand() # Random x and y coordinates between 0 and 1

        # Check if the point lies within the unit circle
        if x^2 + y^2 <= 1
            inside_circle += 1
        end
    end

    # Estimate Pi
    pi_estimate = (inside_circle / num_points) * 4
    return pi_estimate
end

# Example usage
num_points = 1000000 # Number of random points
pi_estimate = monte_carlo_pi_estimation(num_points)
println("Estimated value of Pi: $pi_estimate")

File: C#/MonteCarloPiEstimation.cs
using System;

class MonteCarloPiEstimation
{
    public static double EstimatePi(int numPoints)
    {
        // Monte Carlo Simulation for Pi estimation
        // Args:
        //     numPoints: Number of random points to generate.
        // Returns:
        //     Estimated value of Pi.

        int insideCircle = 0;
        Random rand = new Random();

        for (int i = 0; i < numPoints; i++)
        {
            double x = rand.NextDouble();
            double y = rand.NextDouble();

            // Check if point is inside the unit circle
            if (x * x + y * y <= 1)
            {
                insideCircle++;
            }
        }

        // Calculate Pi estimation
        double piEstimate = (double)insideCircle / numPoints * 4;
        return piEstimate;
    }

    static void Main()
    {
        int numPoints = 1000000; // Number of random points
        double piEstimate = EstimatePi(numPoints);
        Console.WriteLine("Estimated value of Pi: " + piEstimate);
    }
}
